obo/KeyVaultCredentials.cs

```diff
@@ -20,23 +20,30 @@ public Task<(string clientId, string clientSecret)> GetForTenant(string tenantId
     {
         return this.cache.GetOrCreateAsync(tenantId, async entry =>
         {
-            this.logger.LogDebug("attempting to get CLIENT_ID and CLIENT_SECRET from {key}...", this.config.KEYVAULT_URL);
+            try
+            {
+                this.logger.LogDebug("attempting to get CLIENT_ID and CLIENT_SECRET from {key}...", this.config.KEYVAULT_URL);
 
-            var client = new SecretClient(new Uri(this.config.KEYVAULT_URL), this.credential);
-            var clientIdTask = client.GetSecretAsync($"{tenantId}-CLIENT-ID");
-            var clientSecretTask = client.GetSecretAsync($"{tenantId}-CLIENT-SECRET");
-            await Task.WhenAll(clientIdTask, clientSecretTask);
+                var client = new SecretClient(new Uri(this.config.KEYVAULT_URL), this.credential);
+                var clientIdTask = client.GetSecretAsync($"{tenantId}-CLIENT-ID");
+                var clientSecretTask = client.GetSecretAsync($"{tenantId}-CLIENT-SECRET");
+                await Task.WhenAll(clientIdTask, clientSecretTask);
 
-            var clientId = clientIdTask.Result.Value.Value;
-            var clientSecret = clientSecretTask.Result.Value.Value;
+                var clientId = clientIdTask.Result.Value.Value;
+                var clientSecret = clientSecretTask.Result.Value.Value;
 
-            if (string.IsNullOrEmpty(clientId) || string.IsNullOrEmpty(clientSecret))
+                if (string.IsNullOrEmpty(clientId) || string.IsNullOrEmpty(clientSecret))
+                {
+                    throw new HttpException(403, "You are not authorized to use this application.");
+                }
+                this.logger.LogDebug("successfully obtained CLIENT_ID and CLIENT_SECRET from {key}.", this.config.KEYVAULT_URL);
+                return (clientId, clientSecret);
+            }
+            catch (Exception ex)
             {
+                this.logger.LogError(ex, "failed to get CLIENT_ID and CLIENT_SECRET from {key}...", this.config.KEYVAULT_URL);
```

Comment:
There is no point to a try...catch that simply throws the same error. I changed this to log the error and raise a 403.